modin/pandas/dataframe.py

```diff
@@ -2496,18 +2496,22 @@ def __setattr__(self, key, value):
         # - `_siblings`, which Modin initializes before it appears in __dict__
         if key in ["_query_compiler", "_siblings"] or key in self.__dict__:
             pass
-        elif key in self and key not in dir(self):
+        elif key not in dir(self) and key not in self and not is_list_like(value):
+            # Assigning a scalar to a non-existing attribute should not result in a
+            # broadcasted column, but rather just a new attribute.
+            pass
+        elif key not in dir(self):
+            if key not in self:
+                warnings.warn(
```

Comment:
I believe the logic here is leading to test failures in CI, and this was actually a bug we encountered before where we were improperly setting the attributes. If the key is in self and not in dir, we need to call `__setitem__` and return so that we don't call the `object.__setattr__` at the bottom of the call. For example, in the failing test case we have this scenario where we are adding a new column as an attribute:
```python
def test___setattr__not_column():
    pandas_df = pandas.DataFrame([1, 2, 3])
    modin_df = pd.DataFrame([1, 2, 3])
    pandas_df.new_col = [4, 5, 6]
    modin_df.new_col = [4, 5, 6]
    df_equals(modin_df, pandas_df)
```

The current code as is will check if key is not in dir(self), which is true and will also see that key is not in self. We will print the warning, but we will also call `__setitem__` which is not what we want.

Reply:
I think this is outdated now. See below.